app/src/main/java/com/onewelcome/showcaseapp/feature/userderegistration/UserDeregistrationViewModel.kt

```diff
@@ -0,0 +1,78 @@
+package com.onewelcome.showcaseapp.feature.userderegistration
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.michaelbull.result.Err
+import com.github.michaelbull.result.Result
+import com.github.michaelbull.result.onFailure
+import com.github.michaelbull.result.onSuccess
+import com.onegini.mobile.sdk.android.model.entity.UserProfile
+import com.onewelcome.core.usecase.DeregisterUserUseCase
+import com.onewelcome.core.usecase.GetUserProfilesUseCase
+import com.onewelcome.core.usecase.IsSdkInitializedUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class UserDeregistrationViewModel @Inject constructor(
+  private val isSdkInitializedUseCase: IsSdkInitializedUseCase,
+  private val getUserProfilesUseCase: GetUserProfilesUseCase,
+  private val deregisterUserUseCase: DeregisterUserUseCase
+) : ViewModel() {
+
+  var uiState by mutableStateOf(State())
+    private set
+
+  fun onEvent(event: UiEvent) {
+    when (event) {
+      is UiEvent.LoadInitialData -> loadInitialData()
+      is UiEvent.OnUserProfileSelected -> uiState = uiState.copy(selectedUserProfile = event.userProfile)
+      is UiEvent.DeregisterUser -> deregisterUser()
+    }
+  }
+
+  private fun loadInitialData() {
```

Comment:
is this needed? can't be done with viewmodels init {} ?

Reply:
as mentioned above, just another approach to the topic of loading initial data